core/app/models/spree/taxon.rb

```diff
@@ -96,11 +97,19 @@ def set_pretty_name
       end
 
       def name_with_fallback
-        name.blank? ? translated_model.name : name
+        name.blank? ? translated_model['name'] : name
```

Comment:
`['name']` here and in all other places is needed because we need to get the `taxon.name` without the translation, if we call `name` here we will get the translated `name`